src/services/services.ts

```diff
@@ -7708,11 +7708,13 @@ namespace ts {
         function getCodeFixesAtPosition(fileName: string, start: number, end: number, errorCodes: string[]): CodeFix[] {
             synchronizeHostData();
             const sourceFile = getValidSourceFile(fileName);
-
+            const checker = program.getTypeChecker();
             let allFixes: CodeFix[] = [];
 
             forEach(errorCodes, error => {
-                const fixes = codeFixProvider.getFixes(error, sourceFile, start, end);
+                const context = new CodeActionContext(error, sourceFile, { start, length: end - start }, checker);

```

Comment:
seems more Javascripty to use an interface instead of a class for `CodeActionContext` and say

```ts
const context = { error, sourceFile, spec: { start, length: end - start }, checker };
```

since CodeActionContext doesn't use any class features except the constructor.